cmd/fuzz.go

```diff
@@ -0,0 +1,138 @@
+package cmd
+
+import (
+	"context"
+	"github.com/form3tech-oss/go-ci-fuzz/fuzz"
+	"github.com/spf13/cobra"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+const (
+	flagFuzzTime = "fuzz-time"
+	flagFailFast = "fail-fast"
+	flagOut      = "out"
+)
+
+var fuzzCmd = &cobra.Command{
+	Use:   "fuzz [packages...]",
+	Short: "Runs all fuzz targets of packages",
+	Long: `Runs all fuzz targets in <packages> in current directory for the duration of --fuzz-time / N where N is the number of fuzz targets.
+Continues to the next fuzz target on failure unless --fail-fast is defined.
+
+Failing outputs are written to --out directory if specified. The structure is identical to how corpora is stored locally.
+e.g.
+out-dir
+└── testdata
+    └── fuzz
+        └── FuzzTarget
+            └── 0a7e5e215d8c088d4b9c4993d0189a07e81603fbdf64f2ca44738aa27159acef
+`,
+	Run:          fuzzRun,
+	SilenceUsage: true,
+}
+
+func init() {
+	fuzzCmd.Flags().StringP(flagOut, "o", "", "directory to write failing outputs to")
+	fuzzCmd.Flags().Duration(flagFuzzTime, 10*time.Minute, "fuzzing duration for the whole suite")
+	fuzzCmd.Flags().Bool(flagFailFast, false, "exit once failing input is discovered")
+}
+
+func fuzzRun(cmd *cobra.Command, args []string) {
+	ctx := context.Background()
```

Comment:
Here we can use `cmd.Context()` down in `ListFuzzTargets` instead (.. will also make sub routines receptive then to signal)